### DISCUSSION TITLE
How to force timeseries d / differencing parameter to 0

### DISCUSSION THREAD

**Discussion by AlexanderLavelle** [pycaret/pycaret#2077] [Q&A]
I have a mean reverting time series that recently is a bit off due to the pandemic. I want to force Pycaret TimeSeriesExperiment not to difference my time series. I attempted to set in 

```
fit_kwargs={
        'degree':0,
        'd':0,
        'seasonal_order':(0,0,0,12)
}
```

but this did not achieve the desired result (nor were any of my additional fit parameters passed to the models).

How can I set 'Recommended d' or other pieces from the `.setup()`?

**Comment by ngupta23** [comment]
If you are building an ARIMA model, you can hard code the values like this. NOTE: you can not selectively pass only "d" without passing "p" and "q". Hope it helps.

```python
FH = 12
exp = TimeSeriesExperiment()
exp.setup(data=data, fh=FH, session_id=42)

p = 1  # you decide
q = 1  # you decide
d = 0  # difference

P = 0  # you decide
Q = 0  # you decide
D = 0  # Sesasonal Difference
s = 0  # you decide
model = exp.create_model("arima", order=(p, d, q), seasonal_order=(P, D, Q, s))
```

**Reply by AlexanderLavelle** [reply]
@ngupta23 Is there a way to pass pdq, PDQS ranges for hyperparameter tuning?

**Reply by ngupta23** [reply]
Yes, you can pass your own hyperparameter search space using the `custom_grid` argument. I have an example here for a different model, but you can apply the same methodology to ARIMA as well: https://github.com/pycaret/pycaret/discussions/1795
